Shared/Samples/Add features with contingent values/AddFeaturesWithContingentValuesView.AddFeatureView.swift

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension AddFeaturesWithContingentValuesView {
+    /// A view allowing the user to add a feature to the map.
+    struct AddFeatureView: View {
+        /// The view model for the sample.
+        @ObservedObject var model: Model
+        
+        /// The name of the selected status.
+        @State private var selectedStatusName: String?
+        
+        /// The coded value options for the feature's status attribute.
+        @State private var statusOptions: [CodedValue?] = []
+        
+        /// The name of the selected protection.
+        @State private var selectedProtectionName: String?
+        
+        /// The contingent coded value options for the feature's protection attribute.
+        @State private var protectionOptions: [ContingentCodedValue?] = [] {
+            didSet { selectedProtectionName = nil }
+        }
+        
+        /// The selected exclusion area buffer size.
+        @State private var selectedBufferSize: Double?
+        
+        /// The range of size options for the feature's buffer size attribute.
+        @State private var bufferSizeRange: ClosedRange<Double>? {
+            didSet { selectedBufferSize = bufferSizeRange?.lowerBound ?? 0 }
+        }
+        
+        var body: some View {
+            List {
+                Section {
+                    Picker("Status", selection: $selectedStatusName) {
+                        ForEach(statusOptions, id: \.?.name) { option in
+                            Text(option?.name ?? "")
+                        }
+                    }
+                    .onChange(of: selectedStatusName) { newStatusName in
+                        // Update the feature's status attribute.
+                        guard let selectedCodedValue = statusOptions.first(
+                            where: { $0?.name == newStatusName }
+                        ) else { return }
+                        
+                        model.setFeatureAttributeValue(selectedCodedValue?.code, forKey: "Status")
+                        
+                        // Update the protection options.
+                        protectionOptions = model.protectionContingentCodedValues()
+                        protectionOptions.insert(nil, at: 0)
```

Comment:
Nil is needed have the picker start with an empty value option. This mimics the old implementation and helps emphasize that the feature can only be added when criteria is met since it doesn't start off with valid contingencies. I added some comments.